File: sys/osb/bytmov.f
c BYTMOV -- Byte move from array "a" to array "b".  The move must be
c nondestructive, allowing a byte array to be shifted left or right a
c few bytes, hence calls to zlocva() are required to get the addresses of 
c the arrays.

	subroutine bytmov (a, aoff, b, boff, nbytes)

	character*1 a(*), b(*)
	integer	aoff, boff, nbytes
	integer	fwaa, lwaa, fwab, i

		call zlocva (a(aoff), fwaa)
		call zlocva (a(aoff+nbytes-1), lwaa)
		call zlocva (b(boff), fwab)

		if (fwaa .eq. fwab) then
		    return
		else if (fwab .ge. fwaa .and. fwab .le. lwaa) then
		    do 10 i = nbytes-1, 0, -1
			b(boff+i) = a(aoff+i)
 10		    continue
		else
		    do 20 i = 0, nbytes-1
			b(boff+i) = a(aoff+i)
 20		    continue
		endif
	end

File: sys/osb/bytmov.c
/* Copyright(c) 1986 Association of Universities for Research in Astronomy Inc.
 */

#define	import_spp
#define import_knames
#include <iraf.h>

/* BYTMOV -- Byte move from array "a" to array "b".  The move must be
 * nondestructive, allowing a byte array to be shifted left or right a
 * few bytes, hence comparison of the addresses of the arrays is necessary
 * to determine if they overlap.
 * [Specially optimized version for Sun/IRAF].
 */
BYTMOV (
  XCHAR	*a,			/* input byte array			*/
  XINT	*aoff,			/* first byte in A to be moved		*/
  XCHAR	*b,			/* output byte array			*/
  XINT	*boff,			/* first byte in B to be written	*/
  XINT	*nbytes 		/* number of bytes to move		*/
)
{
	if ((a + *aoff) != (b + *boff))
	    memmove ((char *)b + (*boff-1), (char *)a + (*aoff-1), *nbytes);
}
